src/Core/src/Handlers/View/ViewHandler.cs

```diff
@@ -15,35 +15,113 @@
 
 namespace Microsoft.Maui.Handlers
 {
-	public partial class ViewHandler
+	public abstract partial class ViewHandler : IViewHandler
 	{
 		public static PropertyMapper<IView> ViewMapper = new PropertyMapper<IView>
 		{
 			[nameof(IView.BackgroundColor)] = MapBackgroundColor,
 			[nameof(IView.Frame)] = MapFrame,
 			[nameof(IView.IsEnabled)] = MapIsEnabled,
-			[nameof(IView.AutomationId)] = MapAutomationId
+			[nameof(IView.AutomationId)] = MapAutomationId,
+			[nameof(IView.Semantics)] = MapSemantics
 		};
 
+		internal ViewHandler() {}
+
+		bool _hasContainer;
+
+		public bool HasContainer
+		{
+			get => _hasContainer;
+			set
+			{
+				if (_hasContainer == value)
+					return;
+
+				_hasContainer = value;
+
+				if (value)
+					SetupContainer();
+				else
+					RemoveContainer();
+			}
+		}
+
+		protected abstract void SetupContainer();
+
+		protected abstract void RemoveContainer();
+
+		public IMauiContext? MauiContext { get; private set; }
+
+		public NativeView? NativeView
+		{
+			get;
+			private protected set;
+		}
+
+		public IView? VirtualView 
+		{ 
+			get; 
+			private protected set; 
+		}
+
+		public void SetMauiContext(IMauiContext mauiContext) => MauiContext = mauiContext;
+
+		public abstract void SetVirtualView(IView view);
+
+		public abstract void UpdateValue(string property);
+
+		void IViewHandler.DisconnectHandler() => DisconnectHandler(NativeView);
+
+		public abstract Size GetDesiredSize(double widthConstraint, double heightConstraint);
+
+		public abstract void SetFrame(Rectangle frame);
+
+		private protected void ConnectHandler(NativeView? nativeView)
+		{
+		}
+
+		private protected void DisconnectHandler(NativeView? nativeView)
+		{
+#if MONOANDROID
+			if (nativeView.IsAlive() && AccessibilityDelegate != null)
+			{
+				AccessibilityDelegate.Handler = null;
+				AndroidX.Core.View.ViewCompat.SetAccessibilityDelegate(nativeView, null);
+				AccessibilityDelegate = null;
+			}
+#endif
+			if (VirtualView != null)
+				VirtualView.Handler = null;
+
+			VirtualView = null;
+		}
+
 		public static void MapFrame(IViewHandler handler, IView view)
 		{
 			handler.SetFrame(view.Frame);
 		}
 
 		public static void MapIsEnabled(IViewHandler handler, IView view)
 		{
-			(handler.NativeView as NativeView)?.UpdateIsEnabled(view);
+			handler.NativeView?.UpdateIsEnabled(view);
 		}
 
 		public static void MapBackgroundColor(IViewHandler handler, IView view)
 		{
-			(handler.NativeView as NativeView)?.UpdateBackgroundColor(view);
+			handler.NativeView?.UpdateBackgroundColor(view);
 		}
 
 		public static void MapAutomationId(IViewHandler handler, IView view)
 		{
-			(handler.NativeView as NativeView)?.UpdateAutomationId(view);
+			handler.NativeView?.UpdateAutomationId(view);
 		}
 
+#if !MONOANDROID
+		public static void MapSemantics(IViewHandler handler, IView view)
+		{
+			handler.NativeView?.UpdateSemantics(view);
+		}
+#endif
```

Comment:
Maybe this 1 line can be copied into the 2 platform files?

Reply:
done